### PR TITLE
Automate creating GH releases from GitLab

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**

### CODE REVIEW

File: scripts/release.sh [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    "https://uploads.github.com/repos/signalfx/splunk-otel-java/releases/${gh_release_id}/assets?name=$(basename $artifact)" > /dev/null
+}
+
+build_project
+gh_release_id=$(create_release)
+
+for f in dist/*
+do
+  upload_artifact "$gh_release_id" "$f"
+done
```
Comment (iNikem) [inline scripts/release.sh:76]: are you sure you don't want to use existing `softprops/action-gh-release` action for that?

Comment (mateuszrzeszutek) [reply scripts/release.sh:76]: As it is now we can't do signing in GH, so I can't...

Comment (iNikem) [reply scripts/release.sh:76]: I don't see any mentioning of signing in that action description 🤔

File: scripts/release.sh [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  mkdir -p ~/.gnupg
+  chmod 700 ~/.gnupg
+  cat > ~/.gnupg/gpg.conf <<EOF
+no-tty
+pinentry-mode loopback
+EOF
+
+  echo ">>> Importing secret key ..."
+  echo "$GPG_SECRET_KEY" > seckey.gpg
+  trap "rm seckey.gpg" EXIT
```
Comment (iNikem) [inline scripts/release.sh:36]: This will not remove file if script gets killed. Use more signals, at least `KILL`

File: scripts/release.sh [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  gpg --batch --allow-secret-key-import --import seckey.gpg
+}
+
+build_project() {
+  mkdir -p dist
+
+  echo ">>> Building the javaagent ..."
+  ./gradlew -Prelease.useLastTag=true build final closeAndReleaseSonatypeStagingRepository -x test --no-daemon
```
Comment (iNikem) [inline scripts/release.sh:44]: Why no tests? How can we be sure that this given commit is not broken?

File: scripts/release.sh (hunk 2 above)
Comment (mateuszrzeszutek) [reply scripts/release.sh:36]: Good point, will add 👍

File: scripts/release.sh (hunk 3 above)
Comment (mateuszrzeszutek) [reply scripts/release.sh:44]: Test are run in the `build` stage/job - see `.gitlab-ci.yml`. This script is only run once the build passes.

Comment (mateuszrzeszutek) [reply scripts/release.sh:44]: Well, except smoke tests - gitlab runners can't into docker containers, so we only run those on GitHub.

File: scripts/release.sh [hunk 4]
```diff
@@ -33,7 +33,7 @@ EOF
 
   echo ">>> Importing secret key ..."
   echo "$GPG_SECRET_KEY" > seckey.gpg
-  trap "rm seckey.gpg" EXIT
+  trap "rm seckey.gpg" EXIT KILL
```
Comment (iNikem) [inline scripts/release.sh:36]: INT, TERM, STOP?

Comment (mateuszrzeszutek) [reply scripts/release.sh]: Done

File: agent/build.gradle.kts [hunk 5]
```diff
@@ -138,7 +138,7 @@ tasks {
             developer {
               id.set("splunk")
               name.set("Splunk Instrumentation Authors")
-              email.set("support+sonatype@signalfx.com")
+              email.set("support+java@signalfx.com")
```
Comment (jkwatson) [inline agent/build.gradle.kts:141]: is this a thing? Who get these emails, anyway?

Comment (mateuszrzeszutek) [reply agent/build.gradle.kts:141]: No idea 🤷‍♂️ 
`support+java` was what was in the signalfx-java repo, so I wanted to use the same value. And I used it as the GPG key name too.

**Review by iNikem — APPROVED** [review]

File: agent/build.gradle.kts (hunk 5 above)
Comment (jkwatson) [reply agent/build.gradle.kts:141]: I'll put in a quick PR to change Android to use this as well

Comment (mateuszrzeszutek) [reply agent/build.gradle.kts:141]: 👍 

I'm doing the same thing for other repos too

**Pull request merged**